aws/resource_aws_cognito_user.go

```diff
@@ -0,0 +1,284 @@
+				Username:       aws.String(d.Get("username").(string)),
+				UserPoolId:     aws.String(d.Get("user_pool_id").(string)),
+				UserAttributes: expandCognitoUserAttributes(upd),
+			}
+			_, err := conn.AdminUpdateUserAttributes(params)
+			if err != nil {
+				if isAWSErr(err, "ResourceNotFoundException", "") {
+					log.Printf("[WARN] Cognito User %s is already gone", d.Id())
+					d.SetId("")
+					return nil
+				}
+				return fmt.Errorf("Error updating Cognito User Attributes: %s", err)
+			}
+		}
+		if len(del) > 0 {
+			params := &cognitoidentityprovider.AdminDeleteUserAttributesInput{
+				Username:           aws.String(d.Get("username").(string)),
+				UserPoolId:         aws.String(d.Get("user_pool_id").(string)),
+				UserAttributeNames: del,
+			}
+			_, err := conn.AdminDeleteUserAttributes(params)
+			if err != nil {
+				if isAWSErr(err, "ResourceNotFoundException", "") {
+					log.Printf("[WARN] Cognito User %s is already gone", d.Id())
+					d.SetId("")
+					return nil
+				}
+				return fmt.Errorf("Error updating Cognito User Attributes: %s", err)
+			}
+		}
+	}
+
+	return resourceAwsCognitoUserRead(d, meta)
+}
+
+func resourceAwsCognitoUserDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cognitoidpconn
+
+	params := &cognitoidentityprovider.AdminDeleteUserInput{
+		Username:   aws.String(d.Get("username").(string)),
+		UserPoolId: aws.String(d.Get("user_pool_id").(string)),
+	}
+
+	log.Print("[DEBUG] Deleting Cognito User")
+
+	_, err := conn.AdminDeleteUser(params)
+	if err != nil {
+		return fmt.Errorf("Error deleting Cognito User: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsCognitoUserImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	idSplit := strings.Split(d.Id(), "/")
+	if len(idSplit) != 2 {
+		return nil, errors.New("Error importing Cognito User. Must specify user_pool_id/username")
+	}
+	userPoolId := idSplit[0]
+	name := idSplit[1]
+	d.Set("user_pool_id", userPoolId)
+	d.Set("username", name)
+	return []*schema.ResourceData{d}, nil
+}
+
+func expandCognitoUserAttributes(tfSet *schema.Set) []*cognitoidentityprovider.AttributeType {
+	if tfSet.Len() == 0 {
+		return nil
+	}
+
+	apiList := make([]*cognitoidentityprovider.AttributeType, 0, tfSet.Len())
+
+	for _, tfAttribute := range tfSet.List() {
+		apiAttribute := tfAttribute.(map[string]interface{})
+		apiList = append(apiList, &cognitoidentityprovider.AttributeType{
+			Name:  aws.String(apiAttribute["name"].(string)),
+			Value: aws.String(apiAttribute["value"].(string)),
+		})
+	}
+
+	return apiList
+}
+
+func flattenCognitoUserAttributes(apiList []*cognitoidentityprovider.AttributeType) *schema.Set {
+	if len(apiList) == 1 {
+		return nil
+	}
+
+	tfList := []interface{}{}
+
+	for _, apiAttribute := range apiList {
+		// not sure if this is the best way to deal with this system attrubute
+		if *apiAttribute.Name == "sub" {
```

Comment:
I'm not really sure if this is the best way to remove `sub` attribute from the resource. 

From [aws docs](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html):
> The SignUp API generates a persistent UUID for your user, and uses it as the immutable username attribute internally. This UUID has the same value as the sub claim in the user identity token.

So every user has this attribute but it cannot be modified or deleted.